ic-stable-structures/src/memory_mapped_files/memory.rs

```diff
@@ -12,56 +15,87 @@ const WASM_PAGE_SIZE_IN_BYTES: u64 = 65536;
 pub struct MemoryMappedFileMemoryManager {
     base_path: PathBuf,
     is_persistent: bool,
+    created_memory_resources: RwLock<BTreeMap<PathBuf, MemoryMappedFileMemory>>,
 }
 
 impl MemoryMappedFileMemoryManager {
     pub fn new(base_path: PathBuf, is_persistent: bool) -> Self {
         Self {
             base_path,
             is_persistent,
+            created_memory_resources: Default::default(),
+        }
+    }
+
+    pub fn flush_and_save_copies_to(&self, path: impl AsRef<Path>) -> Result<(), MemMapError> {
+        let created_memory_resources = self.created_memory_resources.read();
```

Comment:
This implementation allows changing other memory files while the copy one memory is happening. Is this intentional? How do we guarantee consistency while copying?

Reply:
That's not true, there are many patterns to lock all the internal memories or to reject write operations while copying

Reply:
I don't think we should lock on the level of the memory. Because without lock on the state level this lock would be useless - we may start some state operation, do `write_1`, release lock and start creating backup when having some intermediate state.